### DISCUSSION TITLE
Option to disable self managing displayname, name, and legal name

### DISCUSSION THREAD

**Discussion by scaredmushroom** [kanidm/kanidm#2770] [Q&A]
We have the need that usernames have to be in a specific format and also not be changed w/o approval.

Reasons include that we want to circumvented that students choose unprofessional names ("trolling") and be recognizable to other students and tutors during exercises (especially while working on lab servers – UUIDs are rather impractical here for groupwork).

Hence I wanted to ask if it is somehow possible to disable self managing displaynames, names, and legal names, so that they can be changed on request? If not I also would be thankful for any ideas on how to deal with this problem.

**Comment by Firstyear** [comment]
See #2772
